server/services/fileManager.ts

```diff
@@ -0,0 +1,34 @@
+import b2 from 'backblaze-b2';
+import fs from 'fs';
+
+export default class FileManager {
+  protected b2: b2;
+
+  constructor() {
+    this.b2 = new b2({
+      applicationKeyId: process.env.BACKBLAZE_APP_KEY_ID as string,
+      applicationKey: process.env.BACKBLAZE_APP_KEY as string,
+    });
+    this.b2.authorize();
+  }
+  /**
+   * Upload the file.
+   *
+   * @param localFilePath - Path to the file on the temporary directory.
+   * @param fileName	- The name of the file.
+   * @return The location of the file on the file storage service
+   */
+  async upload(localFilePath: string, fileName: string) {
+    const uploadUrl = await this.b2.getUploadUrl({
+      bucketId: process.env.BACKBLAZE_BUCKET_ID as string,
+    });
+    await this.b2.uploadFile({
+      fileName: fileName,
+      data: fs.readFileSync(localFilePath),
+      uploadUrl: uploadUrl.data.uploadUrl,
+      uploadAuthToken: uploadUrl.data.authorizationToken,
+    });
+
+    return `https://f004.backblazeb2.com/file/${process.env.BACKBLAZE_BUCKET_NAME}/${fileName}`;
```

Comment:
@plausibly Can you add the appropriate env variables to heroku please?

Reply:
Sure, just DM me what the variable should be equal to

Reply:
I used my work credentials to test it. Can you sign up for an account? It's also easier to have one person maintain all the credentials. You don't need to provide payment details.